code/datums/abilities/critter/eat_limb.dm

```diff
@@ -20,30 +21,57 @@
 		var/datum/targetable/critter/stomach_retreat/stomach_abil = src.holder.getAbility(/datum/targetable/critter/stomach_retreat)
 		if (stomach_abil?.inside)
 			return TRUE
+		if (ishuman(target))
+			var/mob/living/carbon/human/targetHuman = target
+			var/list/randLimbBase = list("r_arm", "r_leg", "l_arm", "l_leg")
+			var/list/randLimb
+			for (var/potential_limb in randLimbBase) // build a list of limbs the target actually has
+				if (targetHuman.limbs.get_limb(potential_limb))
+					LAZYLISTADD(randLimb, potential_limb)
+			src.torn_limb = targetHuman.limbs.get_limb(pick(randLimb))
 		if (ishuman(target) || istype(target, /obj/item/parts/human_parts))
-			src.holder.owner.visible_message(SPAN_ALERT("<b>[holder.owner] starts to gnaw at [target]!</b>"))
+			src.holder.owner.visible_message(SPAN_ALERT("<b>[holder.owner] leaps, latching onto and gnawing at [src.torn_limb]!</b>"))
 		else
 			return
-		actions.start(new/datum/action/bar/icon/eat_limb(target, holder.owner), holder.owner)
+		actions.start(new/datum/action/bar/icon/eat_limb(target, holder.owner, src.torn_limb), holder.owner)
 
 /datum/action/bar/icon/eat_limb
 	duration = 1 SECONDS
-	interrupt_flags = INTERRUPT_MOVE | INTERRUPT_ACT | INTERRUPT_ACTION
 	icon = 'icons/mob/screen1.dmi'
 	icon_state = "grabbed"
+	interrupt_flags = null
 	var/atom/target
 	var/mob/user
+	var/datum/human_limbs/tornlimb
 
-	New(Target, User)
+	New(Target, User, Tornlimb)
 		target = Target
 		user = User
+		tornlimb = Tornlimb
 		..()
 
 	onStart()
 		..()
 		if (ishuman(target))
 			src.duration = 5 SECONDS
+			APPLY_ATOM_PROPERTY(user, PROP_MOB_CANTMOVE, "chomping")
+			var/mob/living/carbon/human/human = target
+			user.set_loc(human)
+			human.vis_contents += user
+			if (istype(tornlimb, /obj/item/parts/human_parts/leg/))
```

Comment:
Don't use types to check the side, use slots. What if they have a non-human arm?